keras_cv/layers/preprocessing/random_rotation_test.py

```diff
@@ -99,3 +99,33 @@ def test_ragged_bounding_boxes(self):
             ragged_rank=1,
         )
         self.assertAllClose(expected_output, output["bounding_boxes"])
+
+    def test_augment_sparse_segmentation_mask(self):
+        classes = 8
+
+        input_images = np.random.random((2, 20, 20, 3)).astype(np.float32)
+        masks = np.random.randint(classes, size=(2, 20, 20, 1))
+        inputs = {"images": input_images, "segmentation_masks": masks}
+
+        # 90 rotation.
+        layer = RandomRotation(factor=(0.25, 0.25), segmentation_classes=classes)
+        outputs = layer(inputs)
+
+        expected_masks = np.rot90(masks, axes=(1, 2))
+
+        self.assertAllClose(expected_masks, outputs["segmentation_masks"])
+
+    def test_augment_one_hot_segmentation_mask(self):
+        classes = 8
+
+        input_images = np.random.random((2, 20, 20, 3)).astype(np.float32)
+        masks = tf.one_hot(np.random.randint(classes, size=(2, 20, 20)), classes)
+        inputs = {"images": input_images, "segmentation_masks": masks}
+
+        # 90 rotation.
```

Comment:
We should do strict validation about non 90' rotation, since that will trigger the mix of pixel and argmax.

Reply:
Good call -- I've added some assertions to this test case which verify that, for a 45-degree rotation, no interpolation takes place.